iroh-rpc-client/src/client.rs

```diff
@@ -181,9 +181,9 @@ impl Client {
             }
 
             let mut stream = futures::stream::select_all(streams);
-            while let Some(status) = stream.next().await {
-                status_table.update(status).unwrap();
-                yield status_table.clone();
+            while let Some(s) = stream.next().await {
+                status.update(s).unwrap();
```

Comment:
So this will panic if the service is unknown, right? Could the ClientStatus also simply be a map from service name to status, then you would not have to hardcode the service names?

Reply:
No, actually this is fine. It is better for the code using this to have the three services we care about as explicit fields.

Just maybe make this an expect so we have a clue what is going on if this panics.

Reply:
this `unwrap` has been refactored away